homework/index.js

```diff
@@ -29,23 +31,136 @@
+    );
+    createElement(
+      ul,
+      'h3',
+      'span',
+      { text: ' Updated:  ' },
+      { text: dateChange(repo.updated_at) },
+    );
+  }
+
+  function renderContributes(repositories, ul) {
+    const url = repositories.contributors_url;
+    fetchJSON(url)
+      .then(contributions => {
+        ul.innerHTML = ' ';
+        contributions.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const infoImg = createAndAppend('div', li, { class: 'infoImg' });
+          createAndAppend('img', infoImg, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+          });
+          const infoName = createAndAppend('div', li, { class: 'infoName' });
+          createAndAppend('a', infoName, {
+            text: contributor.login,
+            href: contributor.html_url,
+            target: '_blank',
+          });
+          createAndAppend('span', infoName, {
+            text: contributor.contributions,
+            class: 'number',
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', root, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    fetchJSON(url)
+      .then(repos => {
+        const root = document.getElementById('root');
+        const header = createAndAppend('header', root, {
+          text: 'HYF Repositories',
+          class: 'title',
+        });
+        const select = createAndAppend('select', header, {
+          class: 'selection',
+        });
+        repos
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .forEach((repo, index) =>
+            createAndAppend('option', select, {
+              text: repo.name,
+              value: index,
+            }),
+          );
+        const mainContainer = createAndAppend('main', root, {
+          class: 'main-container',
+        });
+        const repoContainer = createAndAppend('div', mainContainer, {
+          class: 'repo-container',
+        });
+        const ul = createAndAppend('ul', repoContainer, {
+          class: 'listContainer',
+        });
+        renderRepoDetails(repos[0], ul);
+        const contribContainer = createAndAppend('div', mainContainer, {
+          class: 'contributors-container',
+        });
+        createAndAppend('h3', contribContainer, {
+          text: 'Contributions',
+        });
+        const ulContributes = createAndAppend('ul', contribContainer, {
+          class: 'itemList',
+        });
+        renderContributes(repos[0], ulContributes);
+
+        select.addEventListener('click', () => {
+          const repo = repos[select.value];
+          renderRepoDetails(repo, ul);
+          renderContributes(repo, ulContributes);
+        });
+      })
+      .catch(err => {
         createAndAppend('div', root, {
```

Comment:
See my previous comment on the variable `root`.

Reply:
Thank you for your feedback and advice. i'll check again homework and fix those details.